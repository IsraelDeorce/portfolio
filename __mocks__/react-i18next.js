const mock = jest.createMockFromModule('react-i18next')

const useTranslation = () => ({
  t: (key) => key
})

mock.useTranslation = useTranslation

module.exports = mock
