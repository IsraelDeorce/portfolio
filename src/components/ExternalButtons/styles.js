import styled from 'styled-components'

export const ExternalButtons = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
  font-size: 1.25rem;
`
