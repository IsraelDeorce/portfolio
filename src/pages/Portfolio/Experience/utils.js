import { SiHp, SiDell } from 'react-icons/si'
import { GiClapperboard, GiProgression } from 'react-icons/gi'
import pucrs from '../../../assets/pucrs.svg'
import ages from '../../../assets/ages.jpg'
export const projects = [
  {
    image: <SiHp color='#0096D6' />,
    title: 'HP Inc.',
    subtitle: 'subtitle',
    description: 'desc.hp'
  },
  {
    image: <SiDell color='#007DB8' />,
    title: 'Dell Inc.',
    subtitle: 'subtitle',
    description: 'desc.dell'
  },
  {
    image: <img src={pucrs} />,
    title: 'PUCRS',
    subtitle: 'desc.research_intern',
    description: 'desc.intern'
  },
  {
    image: <GiClapperboard color='#FF7F00' />,
    title: 'Gina Tells',
    subtitle: 'desc.personal_project',
    description: 'desc.ginatells'
  },
  {
    image: <GiProgression  color='#0E6B0E' />,
    title: 'Software KPIs',
    subtitle: 'desc.personal_project',
    description: 'desc.kpis'
  },
  {
    image: <img src={ages} />,
    title: 'AGES',
    subtitle: 'desc.nonprofit_engineer',
    description: 'desc.ages'
  }
]
