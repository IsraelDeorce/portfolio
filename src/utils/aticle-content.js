const articles = [
    {
        name: 'learn-react',
        title: 'The Fastest Way to Learn React',
        content: ['blablabla', 'blobloblo']
    },
    {
        name: 'learn-react2',
        title: 'The Fastest Way to Learn React2',
        content: ['blablabla2', 'blobloblo2']
    },
    {
        name: 'learn-react3',
        title: 'The Fastest Way to Learn React3',
        content: ['blablabla3', 'blobloblo3']
    }
]

export default articles