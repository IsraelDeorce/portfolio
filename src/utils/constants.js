export const LINKS = {
  GDRIVE: 'https://docs.google.com/document/d/1gwsxihd7lOVGqM2HzhD2VqOYW85ExDR7hLntG_dOQyg/edit?usp=sharing',
  WHATSAPP: 'https://wa.me/5551980254545',
  GITHUB: 'https://github.com/IsraelDeorce',
  LINKEDIN: 'https://www.linkedin.com/in/israeldeorce',
}

export const TEXT = {
  MYNAME: 'Israel Deorce',
  EMAIL: 'israel.idvj@gmail.com',
  PHONE: '+55 (51) 98025-4545',
  PORTO: 'Porto Alegre'
}
