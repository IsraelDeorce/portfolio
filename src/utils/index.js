export function formatLanguage (language) {
  if (typeof language === 'string') {
    if (language.length > 2) language = language.substring(0, 2)
    language = language.toLowerCase()
  }
  return language
}
