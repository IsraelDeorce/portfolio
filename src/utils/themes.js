const DarkTheme = {
  c_background: '#1E1E1E',
  c_menu: '#252526',
  c_tab: '#3C3C3C',
  c_string: '#CE9078',
  c_function: '#DCDCAA',
  c_varType: '#569CD6',
  c_varDeclaration: '#4FC1DB',
  c_varProperty: '#9CDCFE',
  c_brackets: '#D4D4C8',
  c_component: '#4EC9B0',
  c_number: '#A7CEA8',
  c_importExport: '#C586C0'
}

const LightTheme = {
  c_background: '#FFFFFF',
  c_menu: '#F3F3F3',
  c_tab: '#DDDDDD',
  c_string: '#A31515',
  c_function: '#795D26',
  c_varType: '#0000E1',
  c_varDeclaration: '#0071C1',
  c_varProperty: '#000F80',
  c_brackets: '#000000',
  c_component: '#267F5A',
  c_number: '#098658',
  c_importExport: '#AF00DB'
}

const Themes = {
  dark: DarkTheme,
  light: LightTheme
}

export default Themes
