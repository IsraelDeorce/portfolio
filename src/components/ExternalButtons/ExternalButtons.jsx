import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import { FaGoogleDrive } from 'react-icons/fa'
import { UIButton } from '../'
import { LINKS } from '../../utils/constants'
import * as S from './styles'

function ExternalButtons() {
  const { t } = useTranslation()
  const navigate = useNavigate()

    const handleClick = () => {
      navigate('blog')
    }

  return (
    <S.ExternalButtons>
      <UIButton id='blog-button' label={t('button.visit_my_blog')} onClick={handleClick} />
      <UIButton
          id='resume-button'
          onClick={() => window.open(LINKS.GDRIVE, '_blank')}
          secondary
          label={t('button.resume')}
          rightIcon={<FaGoogleDrive />}
        />
    </S.ExternalButtons>
  )
}

export default ExternalButtons
