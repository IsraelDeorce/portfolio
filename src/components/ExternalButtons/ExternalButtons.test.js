import { t } from 'react-i18next'
import { FaGoogleDrive } from 'react-icons/fa'
import ExternalButtons from './ExternalButtons'
import { LINKS } from '../../utils/constants'

jest.mock('react-router-dom')

describe('<ExternalButtons/>', () => {
  afterEach(() => jest.clearAllMocks())

  it('renders a Visit my Blog UIButton with correct label and translation', () => {
    expect(shallow(<ExternalButtons/>).find('UIButton#blog-button').first().prop('label')).toBe('button.visit_my_blog')
    expect(t).toHaveBeenCalledWith('button.visit_my_blog')
  })

  describe('Resume Button', () => {
    it('renders a Resume UIButton with correct label and translation', () => {
      expect(shallow(<ExternalButtons/>).find('UIButton#resume-button').prop('label')).toBe('button.resume')
      expect(t).toHaveBeenCalledWith('button.resume')
    })
  
    it('renders a Resume UIButton with right icon as google drive', () => {
      expect(shallow(<ExternalButtons/>).find('UIButton#resume-button').prop('rightIcon')).toStrictEqual(<FaGoogleDrive/>)
    })

    it('opens a new window to the google drive resume', () => {
      global.open = jest.fn()      
      shallow(<ExternalButtons/>).find('UIButton#resume-button').simulate('click')
      expect(global.open).toHaveBeenCalledTimes(1)
      expect(global.open).toHaveBeenCalledWith(LINKS.GDRIVE, '_blank')
    })
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<ExternalButtons />)).toMatchSnapshot()
    })
  })
})
