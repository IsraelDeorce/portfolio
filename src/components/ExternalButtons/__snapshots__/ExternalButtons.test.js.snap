// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalButtons/> Snapshot renders without crashing using shallow 1`] = `
<styled.div>
  <UIButton
    id="blog-button"
    label="button.visit_my_blog"
    onClick={[Function]}
  />
  <UIButton
    id="resume-button"
    label="button.resume"
    onClick={[Function]}
    rightIcon={<FaGoogleDrive />}
    secondary={true}
  />
</styled.div>
`;
